src/gui/activitywidget.cpp

```diff
@@ -400,6 +214,154 @@ void ActivityWidget::slotOpenFile(QModelIndex indx)
+            int indx = host.indexOf(QChar('@'));
+            if( indx>-1 ) {
+                host.remove(0, 1+indx);
+            }
+            if( !host.isEmpty() ) {
+                if( accNotified.contains(host)) {
+                    accNotified[host] = accNotified[host]+1;
+                } else {
+                    accNotified[host] = 1;
+                }
+            }
+            _guiLoggedNotifications.insert(activity._id);
+        }
+    }
+
+    _ui->_notifyLabel->setHidden( _widgetForNotifId.isEmpty() );
+    _ui->_notifyScroll->setHidden( _widgetForNotifId.isEmpty() );
+
+    int newGuiLogCount = accNotified.count();
+
+    if( newGuiLogCount > 0 ) {
+        // restart the gui log timer now that we show a notification
+        _guiLogTimer.restart();
+
+        // Assemble a tray notification
+        QString msg = tr("You received %n new notification(s) from %2.", "", accNotified[accNotified.keys().at(0)]).
+                arg(accNotified.keys().at(0));
+
+        if( newGuiLogCount >= 2 ) {
+            QString acc1 = accNotified.keys().at(0);
+            QString acc2 = accNotified.keys().at(1);
+            if( newGuiLogCount == 2 ) {
+                int notiCount = accNotified[ acc1 ] + accNotified[ acc2 ];
+                msg = tr("You received %1 new notifications from %2 and %3.").arg(notiCount).arg(acc1).arg(acc2);
+            } else {
+                msg = tr("You received new notifications from %1, %2 and other accounts.").arg(acc1).arg(acc2);
+            }
+        }
+
+        emit guiLog(Theme::instance()->appNameGUI() + QLatin1String(" ") + tr("Notifications - Action Required"),
+                    msg);
+    }
+}
+
+void ActivityWidget::slotSendNotificationRequest(const QString& accountName, const QString& link, const QString& verb)
+{
+    qDebug() << Q_FUNC_INFO << "Server Notification Request " << verb << link << "on account" << accountName;
+    NotificationWidget *theSender = qobject_cast<NotificationWidget*>(sender());
+
+    const QStringList validVerbs = QStringList() << "GET" << "PUT" << "POST" << "DELETE";
+
+    if( validVerbs.contains(verb)) {
+        AccountStatePtr acc = AccountManager::instance()->account(accountName);
+        if( acc ) {
+            NotificationConfirmJob *job = new NotificationConfirmJob(acc->account());
+            QUrl l(link);
+            job->setLinkAndVerb(l, verb);
+            job->setWidget(theSender);
+            connect( job, SIGNAL( networkError(QNetworkReply*)),
+                                  this, SLOT(slotNotifyNetworkError(QNetworkReply*)));
+            connect( job, SIGNAL( jobFinished(QString, int)),
+                     this, SLOT(slotNotifyServerFinished(QString, int)) );
+            job->start();
+
+            // count the number of running notification requests. If this member var
+            // is larger than zero, no new fetching of notifications is started
+            _notificationRequests++;
```

Comment:
@ckamm good point. I do not think it is a problem, because: The notification widget is reused if it already existed. The buttons are re-created, if new notifications appear, but the notification job end request does not touch the button pointers, but gets the pointers from the button box.

Theoretically the buttons could change for the same notification. But even than, if the notification was answered, that would not be a problem. 

I conclude that I think it is fine :-)